### PR TITLE
Adding option to write launcher output to file.

### PR DESCRIPTION
**Pull request by paulot**
One common issue that we have at LinkedIn when using Testem is that we have no information on the underlying launcher that is propagated to the user. When running thousands of tests, it is not uncommon that browsers crash. This PR adds a flag "log_launchers" that when set to true will write the launcher's stdout and stderr to a file. This is particularly useful when things crash in ci, as we can have phantomjs' debug information propagated to the developer, rather than swallowed by Testem.

### CODE REVIEW

**Comment by paulot** [comment]
@johanneswuerbach Any concerns over this?

**Comment by johanneswuerbach** [comment]
I like the general idea, but would prefer to write the output into the testem debug log https://github.com/testem/testem/blob/241b1e42e96cd7e4d656efda6475e96e1f7978ae/testem.js#L18 as is already provides path customisation etc.

WDYT?

**Comment by paulot** [comment]
Sounds good, I'm ok with that. How would it look in the testem log though? Would it all be outputted to the same log file?

**Comment by johanneswuerbach** [comment]
I would use the debug flag without log launchers and always log in case of unexpected exit / error.

Would this help or do you need those logs when everything is fine?

**Comment by johanneswuerbach** [comment]
For differentiation between launchers maybe a common prefix per launcher so you can easily cat log | grep launcherId ?

**Comment by paulot** [comment]
Sounds good, logging onProcessError rather than onProcessExit is fine by me.

**Comment by paulot** [comment]
Ok, it doesn't seem like onProcessError catches all of the error use cases that I'm looking for, so it doesn't seem to work. Any suggestions, or should I just keep it in onProcessExit?

**Comment by johanneswuerbach** [comment]
It feels like you want to log in all cases where `toResult` is called https://github.com/testem/testem/blob/master/lib/runners/browser_test_runner.js#L79

Those are usually out-of-order exits when the process dies, the browser never connects or disconnects in the middle.

**Comment by paulot** [comment]
Ok latest commit refactors the piece of code you pointed out and adds the logging functionality to the testem's log. Let me know if you think there are any additional changes.

File: lib/runners/browser_test_runner.js [hunk 1]
```diff
@@ -143,6 +141,14 @@ BrowserTestRunner.prototype = {
     return this.launcher.name;
   },
 
+  reportResults: function(err, code, process) {
+    process = process || this.process;
```
Comment (johanneswuerbach) [inline lib/runners/browser_test_runner.js]: Could you name that differently? `process` is a global in node

**Comment by johanneswuerbach** [comment]
Otherwise this looks pretty great, does that fix your debuggability issues?

File: lib/runners/browser_test_runner.js (hunk 1 above)
Comment (paulot) [inline lib/runners/browser_test_runner.js]: Sure thing

**Comment by paulot** [comment]
Yeah, this fixes the debuggability issues, hopefully we'll be able to root cause it now!

Thanks!

**Comment by johanneswuerbach** [comment]
Perfect, could you squash your commits?

**Comment by paulot** [comment]
Ok squashed and merged.

**Comment by paulot** [comment]
@johanneswuerbach Changes are squashed, good to push?

**Comment by johanneswuerbach** [comment]
Yes, sorry. Thanks for doing that 👍

**Pull request merged**